jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/messages/MessageInputStream.java

```diff
@@ -95,7 +95,9 @@ public int read() throws IOException
     @Override
     public int read(final byte[] b, final int off, final int len) throws IOException
     {
-        return read(ByteBuffer.wrap(b, off, len).flip());
+        ByteBuffer buffer = ByteBuffer.wrap(b, off, len).slice();
```

Comment:
Why slice???

Reply:
Slice creates a new buffer with the capacity set to the limit of the previous buffer.
So the capacity of the buffer will be `len`, and not `b.length`. This is important so we never write more bytes than `len`.